### PR TITLE
Report runtime-only reported Descriptions as dynamic TestDescriptors

### LINKED ISSUE #937: Spock tests with @Unroll are not supported by Vintage engine
Tests are executed but no notifications are received for the following sample code:

```groovy
package q;

import spock.lang.Specification
import spock.lang.Unroll

class CalcSpec extends Specification {
    @Unroll
    void "#a plus #b is equals #c"() {
        println "executed"
        expect:
        a + b == c

        where:
        a | b || c
        1 | 1 || 2
        1 | 2 || 3
    }
}
```

```
WARNING: Runner org.spockframework.runtime.Sputnik on class q.CalcSpec reported event for unknown Description: 1 plus 1 is equals 2(q.CalcSpec). It will be ignored.
```

Checked with M5.

Linked IDEA issue: https://youtrack.jetbrains.com/issue/IDEA-159975

### PR DESCRIPTION
**Pull request by marcphilipp**
Fixes #937.

### Definition of Done

- [X] There are no TODOs left in the code
- [X] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [X] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [X] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests)
- [X] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/meta/API.html)
- [x] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [x] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

**Comment by marcphilipp** [comment]
The `CalcSpec` from #937 now works when run from `ConsoleLauncher`:

```
└─ JUnit Vintage ✔
   └─ q.CalcSpec ✔
      ├─ 1 plus 1 is equals 2 ✔
      ├─ 1 plus 2 is equals 3 ✔
      └─ 2 plus 2 is equals 4 ✔
```

but not from IntelliJ:

screen shot 2017-07-15 at 22 02 17

Looks like the same issue as #756 to me.

@akozlova Is there a YouTrack issue for that?

**Comment by marcphilipp** [comment]
One more detail: Vintage reports "q.CalcSpec" as Type.TEST because its JUnit 4 Description returns true for `isTest()` since it does not have any children.

File: documentation/src/docs/asciidoc/release-notes-5.0.0-M6.adoc [hunk 1]
```diff
@@ -72,7 +72,9 @@ on GitHub.
 
 ===== Bug Fixes
 
-* ❓
+* Added support for `Runners` that report events for tests not part of the `Description`
+  tree such as Spock's `Sputnik` when used with `@Unroll`. Previously, such tests were not
+  reported at all, now they are reported as dynamic tests.
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/release-notes-5.0.0-M6.adoc]: , --> ;

Comment (marcphilipp) [inline documentation/src/docs/asciidoc/release-notes-5.0.0-M6.adoc]: Fixed in db596b2.

**Pull request merged**